### PR TITLE
TELCODOCS-1875: Add scale out to SiteConfig

### PR DESCRIPTION
**Pull request by amolnar-gh**
Issues:

- https://issues.redhat.com/browse/TELCODOCS-1875
- https://issues.redhat.com/browse/ACM-14071

### CODE REVIEW

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+* You have logged in to the hub cluster as a user with `cluster-admin` privileges.
+* If using {ztp}, you have configured your {ztp} environment. To configure your environment, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/edge_computing/ztp-preparing-the-hub-cluster[Preparing the hub cluster for GitOps ZTP].
+* You have the default cluster templates. To get familiar with the default templates, see xref:../../acm_mce_integration/siteconfig/cluster_templates.adoc#default-templates[Default set of templates]
+* You have installed your cluster with the {sco}. To install a cluster with the {sco}, see see xref:../../acm_mce_integration/siteconfig/install-clusters.adoc##install-clusters[Installing {sno} clusters with the {sco}]
+
+[#scale-out-annotation]
+== Adding a worker node
+
+Add a worker node by updating your `ClusterInstall` CR.
```
Comment (mpmaruthu) [inline mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc]:
```suggestion
Add a worker node by updating your `ClusterInstance` CR.
```

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        - name: aci-node-crs-v1
+          namespace: rhacm
+      bmcAddress: "<bmc_address>"
+      bmcCredentialsName:
+        name: "<bmc_credentials_name>"
+      bootMACAddress: "<boot_mac_address>"
+...
+----
+
+For more information about the required and optional fields, see xref:../../rhacm-docs/apis/clusterinstance.json.adoc#clusterinstance[ClusterInstance].
```
Comment (mpmaruthu) [inline mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc:42]: I can't access the referenced link to see required field definition in ClusterInstance CR.

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
++
+[source,yaml]
+----
+spec:
+  ...
+  nodes:
+    - hostName: "<host_name>"
+      role: "worker"
+      templateRefs:
+        - name: aci-node-crs-v1
```
Comment (mpmaruthu) [inline mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc]:
```suggestion
        - name: ai-node-templates-v1
```

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+[source,terminal]
+----
+oc get managedclusterinfo/<managed_cluster_name> -n <managed_cluster_namespace> \
+-o jsonpath='{range .status.nodeList[*]}{.name}{"\t"}{.conditions}{"\t"}{.labels}{"\n"}{end}'
+----
+
++
+[source,terminal]
+----
+oc get agents -n <managed_cluster_namespace> --watch
```
Comment (mpmaruthu) [inline mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc]: In my opinion, --watch is optional to monitor background events which needs to be removed when checking whether the agent created for the node or not.
```suggestion
oc get agents -n <managed_cluster_namespace> 
```

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+----
+
++
+[source,terminal]
+----
+oc get bmh -n <managed_cluster_namespace> --watch
```
Comment (mpmaruthu) [inline mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc]:
```suggestion
oc get bmh -n <managed_cluster_namespace> 
```
the above review comment applies here.

**Review by mpmaruthu — COMMENTED** [review]
Telco RAN QE:
After my initial review, provided nice to have comments. Please check. Thanks!
Otherwise looks good to me.

File: mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc (hunk 2 above)
Comment (amolnar-gh) [reply mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc:42]: It should point to this section: https://github.com/stolostron/rhacm-docs/blob/2.12_stage/apis/clusterinstance.json.adoc#clusterinstance

Comment (mpmaruthu) [reply mce_acm_integration/siteconfig/siteconfig_scale_out_worker_nodes.adoc:42]: Looks good to me.
/Peri M

**Pull request merged**